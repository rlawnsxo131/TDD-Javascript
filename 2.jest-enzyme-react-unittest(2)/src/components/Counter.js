import React from 'react';
import propTypes from 'prop-types';

const Counter = ({ value, onIncrease, onDecrease }) => {
  return (
    <div>
      <h1>카운터</h1>
      <h2>{value}</h2>
      <button onClick={onIncrease}>+</button>
      <button onClick={onDecrease}>-</button>
    </div>
  );
}

Counter.defaultProps = {
  value: 0,
  onIncrease: () => console.warn('onIncrease not defined'),
  onDecrease: () => console.warn('onDecrease not defined')
}
Counter.propTypes = {
  value: propTypes.number.isRequired,
  onIncrease: propTypes.func.isRequired,
  onDecrease: propTypes.func.isRequired
}

export default Counter;