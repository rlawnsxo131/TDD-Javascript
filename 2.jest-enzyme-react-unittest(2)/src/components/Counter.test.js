import React from 'react';
import { shallow } from 'enzyme';
import Counter from './Counter';

describe('Counter', () => {
  let component = null;
  const mockIncrease = jest.fn();
  const mockDecrease = jest.fn();
  
  it('renders correctly', () => {
    component = shallow(<Counter value={700} onIncrease={mockIncrease} onDecrease={mockDecrease}/>);
  });

  it('matches sanpshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('is 700', () => {
    /* 현재는 이렇게하면 안됨
       expect(component.find('h2').at(0).text(), '700'); */
    expect(component.find('h2').at(0).text()).toBe('700');
  });

  it('calls functions', () => {
    const buttons = component.find('button');
    buttons.at(0).simulate('click');
    buttons.at(1).simulate('click');
    expect(mockIncrease.mock.calls.length).toBe(1);
    expect(mockDecrease.mock.calls.length).toBe(1);
  })
});
