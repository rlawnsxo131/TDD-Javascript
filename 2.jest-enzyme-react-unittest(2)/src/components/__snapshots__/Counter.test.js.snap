// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter matches sanpshot 1`] = `
<div>
  <h1>
    카운터
  </h1>
  <h2>
    700
  </h2>
  <button
    onClick={[MockFunction]}
  >
    +
  </button>
  <button
    onClick={[MockFunction]}
  >
    -
  </button>
</div>
`;
